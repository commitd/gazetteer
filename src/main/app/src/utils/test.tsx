import { StylesProvider } from '@material-ui/styles'
import {
  createHistory,
  createMemorySource,
  History,
  LocationProvider,
} from '@reach/router'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { GenerateId } from 'jss'
import React from 'react'
import { ThemeProvider } from '@committed/components'

const generateClassName: GenerateId = (rule, styleSheet) => {
  const prefix =
    styleSheet === undefined || styleSheet.options.classNamePrefix === undefined
      ? ''
      : styleSheet.options.classNamePrefix
  return `${prefix}-${rule.key}`
}

const LightTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="light">{children}</ThemeProvider>
  </StylesProvider>
)

const DarkTheme: React.FC<{ readonly children?: React.ReactNode }> = ({
  children,
}) => (
  <StylesProvider generateClassName={generateClassName}>
    <ThemeProvider choice="dark">{children}</ThemeProvider>
  </StylesProvider>
)

export const renderPlain = render

export const renderLight = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: LightTheme, ...options })

export const renderDark = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: DarkTheme, ...options })

export function renderWithRouter(
  ui: Readonly<React.ReactElement>,
  route = '/',
  options?: Omit<RenderOptions, 'queries'>,
  delegate = renderLight
): RenderResult & { history: History } {
  const history = createHistory(createMemorySource(route))
  return {
    ...delegate(
      <LocationProvider history={history}>{ui}</LocationProvider>,
      options
    ),
    history,
  }
}
// re-export everything
export * from '@testing-library/react'
// override render method
export { renderLight as render }
export { userEvent }
