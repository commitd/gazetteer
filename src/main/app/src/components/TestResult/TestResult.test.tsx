import React from 'react'
import { render, renderDark } from '../../utils/test'
import { TestResult } from '.'

it('renders without error', () => {
  const { asFragment } = render(
    <TestResult
      text="this is a test"
      results={[
        {
          length: 4,
          value: 'test',
          type: 'TEST',
          offset: 10,
        },
      ]}
    />
  )
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without error', () => {
  const { asFragment } = renderDark(
    <TestResult
      text="this is a test"
      results={[
        {
          length: 4,
          value: 'test',
          type: 'TEST',
          offset: 10,
        },
      ]}
    />
  )
  expect(asFragment()).toMatchSnapshot()
})
