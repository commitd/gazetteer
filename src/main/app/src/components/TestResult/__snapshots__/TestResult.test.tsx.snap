// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without error 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Results
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="makeStyles-legend-top"
      >
        <span
          class="WrappedTypography-root WrappedTypography-root"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            >
              <span
                class="makeStyles-root makeStyles-root"
                style="background: rgba(255, 255, 204, 0.7); padding-left: 8px; padding-right: 8px; border-radius: undefined undefined undefined undefined;"
              >
                TEST
              </span>
            </div>
          </div>
        </span>
      </div>
      <div
        class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div>
          <span
            class="WrappedTypography-root WrappedTypography-root makeStyles-text"
          >
            <span
              class="makeStyles-root makeStyles-root"
            >
              <span>
                this is a 
              </span>
              <span
                class="makeStyles-root makeStyles-root"
                style="background: rgba(255, 255, 204, 0.7); padding-left: 8px; padding-right: 8px; border-radius: undefined undefined undefined undefined;"
              >
                test
                <span
                  class="makeStyles-type"
                >
                  TEST
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without error 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root WithStyles(ForwardRef(Typography))-root Component-root Component-root MuiTypography-h1"
    >
      Results
    </h1>
    <div
      class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
    >
      <div
        class="makeStyles-legend-top"
      >
        <span
          class="WrappedTypography-root WrappedTypography-root"
        >
          <div
            class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
          >
            <div
              class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
            >
              <span
                class="makeStyles-root makeStyles-root"
                style="background: rgba(255, 255, 204, 0.7); padding-left: 8px; padding-right: 8px; border-radius: undefined undefined undefined undefined;"
              >
                TEST
              </span>
            </div>
          </div>
        </span>
      </div>
      <div
        class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root WithStyles(ForwardRef(Card))-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div>
          <span
            class="WrappedTypography-root WrappedTypography-root makeStyles-text"
          >
            <span
              class="makeStyles-root makeStyles-root"
            >
              <span>
                this is a 
              </span>
              <span
                class="makeStyles-root makeStyles-root"
                style="background: rgba(255, 255, 204, 0.7); padding-left: 8px; padding-right: 8px; border-radius: undefined undefined undefined undefined;"
              >
                test
                <span
                  class="makeStyles-type"
                >
                  TEST
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
